gravitee-apim-plugin/gravitee-apim-plugin-endpoint/gravitee-apim-plugin-endpoint-mqtt5/src/main/java/io/gravitee/plugin/endpoint/mqtt5/Mqtt5EndpointConnector.java

```diff
@@ -100,18 +135,12 @@ protected Completable subscribe(ExecutionContext ctx) {
                         .messages(
                             Flowable.defer(
                                 () -> {
-                                    Mqtt5RxClient rxClient = prepareMqtt5Client(ctx).identifier(generateSubscriberIdentifier()).buildRx();
+                                    Mqtt5RxClient mqtt5RxClient = mqtt5ClientFactory.getClient(ctx);
                                     String topic = getTopic(ctx);
                                     return RxJavaBridge
                                         .toV3Flowable(
-                                            rxClient
-                                                .connect(prepareMqttConnect().build())
-                                                .flatMapPublisher(
-                                                    mqttConnAck ->
-                                                        prepareMqtt5Subscribe(ctx, rxClient, topic).applySubscribe().map(this::transform)
-                                                )
+                                            prepareMqtt5Subscribe(ctx, mqtt5RxClient, topic).applySubscribe().map(this::transform)
```

Comment:
Since the client is now shared, there is no call to `close` anymore and that's seems pretty normal. However, it appears that the cancellation doesn't get propagated and the client still continue to consume the topic while the subscriber is not there anymore.
Ex: 

1. start an sse request
2. produce a message each second in the topic -> The messages are well received by the sse consumer.
3. Stop the sse request -> warn appear in the gateway's logs

```
08:20:16.329 [com.hivemq.client.mqtt-4-2] [] WARN  c.h.c.i.m.h.p.i.MqttIncomingPublishService - No publish flow registered for MqttStatefulPublish{stateless=MqttPublish{topic=python/mqtt, payload=10byte, qos=AT_MOST_ONCE, retain=false}, packetIdentifier=-1, dup=false, topicAlias=0, subscriptionIdentifiers=[4]}.
08:20:17.335 [com.hivemq.client.mqtt-4-2] [] WARN  c.h.c.i.m.h.p.i.MqttIncomingPublishService - No publish flow registered for MqttStatefulPublish{stateless=MqttPublish{topic=python/mqtt, payload=10byte, qos=AT_MOST_ONCE, retain=false}, packetIdentifier=-1, dup=false, topicAlias=0, subscriptionIdentifiers=[4]}.
08:20:18.338 [com.hivemq.client.mqtt-4-2] [] WARN  c.h.c.i.m.h.p.i.MqttIncomingPublishService - No publish flow registered for MqttStatefulPublish{stateless=MqttPublish{topic=python/mqtt, payload=10byte, qos=AT_MOST_ONCE, retain=false}, packetIdentifier=-1, dup=false, topicAlias=0, subscriptionIdentifiers=[4]}.
08:20:19.343 [com.hivemq.client.mqtt-4-2] [] WARN  c.h.c.i.m.h.p.i.MqttIncomingPublishService - No publish flow registered for MqttStatefulPublish{stateless=MqttPublish{topic=python/mqtt, payload=10byte, qos=AT_MOST_ONCE, retain=false}, packetIdentifier=-1, dup=false, topicAlias=0, subscriptionIdentifiers=[4]}.
08:20:20.343 [com.hivemq.client.mqtt-4-2] [] WARN  c.h.c.i.m.h.p.i.MqttIncomingPublishService - No publish flow registered for MqttStatefulPublish{stateless=MqttPublish{topic=python/mqtt, payload=10byte, qos=AT_MOST_ONCE, retain=false}, packetIdentifier=-1, dup=false, topicAlias=0, subscriptionIdentifiers=[4]}.
08:20:21.348 [com.hivemq.client.mqtt-4-2] [] WARN  c.h.c.i.m.h.p.i.MqttIncomingPublishService - No publish flow registered for MqttStatefulPublish{stateless=MqttPublish{topic=python/mqtt, payload=10byte, qos=AT_MOST_ONCE, retain=false}, packetIdentifier=-1, dup=false, topicAlias=0, subscriptionIdentifiers=[4]}.
...
```

Reply:
Oo yes, maybe something to do with the subscription itself.